source/AI.cpp

```diff
@@ -105,28 +105,19 @@ namespace {
 		return true;
 	}
 
-	// Determine if the ship has any usable weapons.
-	bool IsArmed(const Ship &ship)
-	{
-		for(const Hardpoint &hardpoint : ship.Weapons())
-		{
-			const Weapon *weapon = hardpoint.GetOutfit();
-			if(weapon && !hardpoint.IsAntiMissile())
-			{
-				if(weapon->Ammo() && !ship.OutfitCount(weapon->Ammo()))
-					continue;
-				return true;
-			}
-		}
-		return false;
-	}
-
 	void Deploy(const Ship &ship, bool includingDamaged)
 	{
 		for(const Ship::Bay &bay : ship.Bays())
-			if(bay.ship && (includingDamaged || bay.ship->Health() > .75) &&
-					(!bay.ship->IsYours() || bay.ship->HasDeployOrder()))
-				bay.ship->SetCommands(Command::DEPLOY);
+			if(bay.ship && (includingDamaged || bay.ship->Health() > .75) && bay.ship->Energy() > .75
+					&& (!bay.ship->IsYours() || bay.ship->HasDeployOrder())
+					&& (bay.ship->IsArmed(true) || (!bay.ship->IsArmed() && !bay.ship->IsEnemyInEscortSystem())))
+			{
+				// Refuse to deploy on low energy
+				if(bay.ship->IsEnergyLow())
+					Messages::Add(bay.ship->Attributes().Category() + " " + bay.ship->Name() + " refuses to deploy due insufficient energy.", Messages::Importance::High);
```

Comment:
👎 why is the player being told about NPC fighters that don't deploy?

(Also, one message per ship is a great way to spam the message display.)

If the player can't do anything about the lack of energy, these don't seem like particularly important or actionable messages.

Reply:
Fair I can remove them.  This would happen if a player misconfigured their battery powered fighters.  However, I added this to flightcheck so I can remove the messages.

Reply:
Removed the messages.